### PR TITLE
Fuyu Multi-image interleaved processor

### PR DESCRIPTION
**Pull request by cliangyu**
# What does this PR do?

Fuyu Multi-image interleaved processor. Test example:
```python
from transformers import FuyuProcessor, FuyuForCausalLM
from PIL import Image
import requests
import torch

# load model and processor
model_id = "adept/fuyu-8b"
processor = FuyuProcessor.from_pretrained(model_id)
model = FuyuForCausalLM.from_pretrained(model_id, device_map="cuda:0", torch_dtype=torch.bfloat16)

def convert(list_of_dicts):# Convert to a dictionary of lists
    dict_of_lists = {}
    for d in list_of_dicts:
        for key, value in d.items():
            if key not in dict_of_lists:
                dict_of_lists[key] = []
            dict_of_lists[key].append(value)
    return dict_of_lists

text_prompt1 = "|IMAGESTART| Generate a coco-style caption. |IMAGESTART| Be reminded that the caption should be longer than 2000 words but shorter than 1 million words. \n"
url1 = "https://huggingface.co/adept/fuyu-8b/resolve/main/bus.png"
image1 = Image.open(requests.get(url1, stream=True).raw)

text_prompt2 = "What doesn this chart describe?\n"
url2 = "https://huggingface.co/adept/fuyu-8b/resolve/main/chart.png"
image2 = Image.open(requests.get(url2, stream=True).raw)

test_examples = [
    # {"text": "|IMAGESTART| Generate a coco-style caption. |IMAGESTART| Be reminded that the caption should be longer than 2000 words but shorter than 1 million words. \n", "images": image1}, # should assert error
    {"text": text_prompt1, "images": [image1, image2]}, # normal
    {"text": text_prompt2, "images": [image2 for i in range(40)]}, # should add indicator
    {"text": "|IMAGESTART||IMAGESTART| Generate a coco-style caption. Be reminded that the caption should be longer than 2000 words but shorter than 1 million words. \n", "images": [image1, image2]}, # normal
    {"text": " Generate a coco-style caption. Be reminded that the caption should be longer than 2000 words but shorter than 1 million words. \n|IMAGESTART||IMAGESTART|", "images": [image1, image2]}, # normal
    # {"text": " Generate a coco-style caption. Be reminded that the caption should be longer than 2000 words but shorter than 1 million words.", "images": None}, # no image, we had error with this case
    {"text": None, "images": [image1]}, # no text
    
]
inputs_to_model = processor(**convert(test_examples), return_tensors="pt", truncation=True).to("cuda:0")
```
## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts, @ArthurZucker and @younesbelkada

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Hey! Feel free to ping @molbap once this is ready for review and cis are green or me if you need help with the CIs!

File: src/transformers/models/fuyu/image_processing_fuyu.py [hunk 1]
```diff
@@ -468,22 +465,22 @@ def preprocess(
...
         if do_resize:
             batch_images = [
                 [self.resize(image, size=size, input_data_format=input_data_format) for image in images]
                 for images in batch_images
             ]
 
-        image_sizes = [get_image_size(images[0], channel_dim=input_data_format) for images in batch_images]
-        image_unpadded_heights = [[image_size[0]] for image_size in image_sizes]
-        image_unpadded_widths = [[image_size[1]] for image_size in image_sizes]
+        batch_image_sizes = [[get_image_size(image, channel_dim=input_data_format) for image in images] for images in batch_images]
```
Comment (SeungyounShin) [inline src/transformers/models/fuyu/image_processing_fuyu.py:476]: I believe the issue is not with the PR itself, but when I execute the code from the PR, I encounter a problem with images being excessively padded.

image

Like this

**Comment by SeungyounShin** [comment]
Hi [@cliangyu](https://github.com/cliangyu),

I've developed enhancements for multi-device support in PR #27587, building upon your work. Before proceeding with a new PR, I'd like to discuss integrating these changes with yours. I can submit a PR to your fork or detail the changes here for your review. Please let me know your preferred approach.

**Review by ArthurZucker — COMMENTED** [review]
btw, let's not push the .gitignore, and it might be good to think about processing in the modeling code like what we are doing in #27662 ! 🤗 cc @molbap we can benchmark anyway before anything

**Pull request closed without merging**